### ISSUE TITLE
IntersectionObserverEntry should have a constructor

### ISSUE THREAD

**Issue by domenic** [slightlyoff/IntersectionObserver#33]
IntersectionObserverEntry objects are not exotic and there's no reason to keep their creation magic that I can see. They should have a constructor. I'd suggest one that takes a dictionary

```
dictionary IntersectionObserverEntryInit {
  required DOMHighResTimeStamp time;
  required DOMRectInit rootBounds;
  required DOMRectInit boundingClientRect;
  required DOMRectInit intersectionRect;
  required Element target;
}
```

You could try to add smart defaults instead of making everything required but that doesn't seem worth the trouble.

**Comment by mpb** [comment]
Hm, would JavaScript really want to create IntersectionObserverEntry objects?
MutationRecord (http://www.w3.org/TR/dom/#interface-mutationrecord) doesn't seem to provide one either.
Or is this a newer standard for specs that MutationObserver predates?

**Comment by domenic** [comment]
> Or is this a newer standard for specs that MutationObserver predates?

Mostly this. Having objects be constructed magically by the platform should be reserved for truly special cases.

One example use case is mocking libraries.

**Comment by mpb** [comment]
Sure. And it's preferred to do an initialization dictionary versus listing each parameter?

**Comment by domenic** [comment]
For something with this many properties, none of which are clearly primary, yeah, I think that's the way to go.

**Comment by mpb** [comment]
Closed via #44

**Comment by RByers** [comment]
Unfortunately it looks like we [never implemented the constructor in blink](https://bugs.chromium.org/p/chromium/issues/detail?id=674519) ☹️
